assignments/lambda-classes.js

```diff
@@ -1 +1,88 @@
 // CODE here for your Lambda Classes
+//base class
+class Person {
+  constructor(attr) {
+    this.name = attr.name;
+    this.age = attr.age;
+    this.location = attr.location;
+  }
+
+  speak() {
+    console.log(`Hello my name is ${this.name}, I am from ${this.location}`);
+  }
+}
+class Instructor extends Person {
+  constructor(attrs) {
+    super(attrs);
+    this.specialty = attrs.specialty;
+    this.favLanguage = attrs.favLanguage;
+    this.catchPhrase = attrs.catchPhrase;
+  }
+
+  demo(subject) {
+    console.log(`Today we are learning ${subject}`);
+  }
+
+  grade(student, subject) {
+    console.log(`${this.name} recieves a perfect score on ${subject}`);
+  }
+}
+class Student extends Instructor {
+  constructor(attr) {
+    super(attr);
+    this.previousBackground = attr.previousBackground;
+    this.className = attr.className;
+    this.favSubjects = attr.favSubjects;
+  }
+
+  listSubjects(cb) {
+    setTimeout(function() {
+      cb(this.favSubjects);
+    }, 500);
+  }
+  prAssignment(subject) {
+    console.log(`${this.name} has submitted a PR for ${subject}`);
+  }
+
+  sprintChallenge(str) {
+    console.log(`${this.name} has begun sprint challenge on ${str}`);
+  }
+}
+class Projectmanager extends Instructor {
+  constructor(attr) {
+    super(attr);
+    this.gradClassName = attr.gradClassName;
+    this.favInstructor = attr.favInstructor;
+  }
+
+  standUp(sc) {
+    console.log(`${this.name} announces to ${sc}, @channel standy times!`);
+  }
+
+  debugsCode(obj, sub) {
+    console.log(`${this.name} debugs ${obj.name}'s code on ${sub}`);
+  }
+}
+
+const student1 = new Student({
+  previousBackground: "Python",
+  className: "CS10",
+  favSubjects: "Constructors and prototypes",
+  favLanguage: "Ruby",
+  name: "Anthony",
+  age: 23,
+  location: "Dallas, TX"
+});
+
+const projectManager1 = new Projectmanager({
+  gradClassName: "CS2",
+  favInstructor: "Josh",
+  specialty: "React",
+  catchPhrase: "To infinity anf beyond!",
+  name: "John"
+});
+
+console.log(student1.prAssignment("CSS"));
```

Comment:
A little confusion here with your tests. None of your methods have any returns so when you console log a function that only runs a console log , you are going to get an undefined popping up in your console. I would pick one way or the other. Either set your methods to do a return instead of a console and console log your methods, or leave your methods as just running a console log and just run your methods down here.